RelationalAI/Client.cs

```diff
@@ -50,17 +50,25 @@ public HttpClient HttpClient
             set { _rest.HttpClient = value; }
         }
 
-        public Task<Database> CreateDatabaseAsync(string database, string engine)
+        public async Task<Database> CreateDatabaseAsync(string database, string engine = null, bool overwrite = false, string source = null)
         {
-            return CreateDatabaseAsync(database, engine, false);
-        }
+            if (engine != null)
+            {
+                return await CreateDatabaseV1Async(database, engine, overwrite);
+            }
 
-        public async Task<Database> CreateDatabaseAsync(string database, string engine, bool overwrite)
-        {
-            var mode = CreateMode(null, overwrite);
-            var transaction = new Transaction(_context.Region, database, engine, mode);
-            await _rest.PostAsync(MakeUrl(PathTransaction), transaction.Payload(null), null, transaction.QueryParams());
-            return await GetDatabaseAsync(database);
+            var data = new Dictionary<string, string>
+            {
+                { "name", database }
+            };
+
+            if (source != null)
+            {
+                data.Add("source_name", source);
+            }
+
+            string rsp = await _rest.PutAsync(MakeUrl(PathDatabase), data) as string;
```

Comment:
Let's discuss tomorrow.. I want to understand the contract in play here... why is the new usage PUT instead of POST for the substantially same behavior? Probably the answer is just that this is the contract of database endpoint but would like to discuss it